### ISSUE TITLE
False positive on "There should be no unused return values for Enum functions" with anonymous function usage

### ISSUE THREAD

**Issue by byronanderson** [rrrene/credo#1167]
### Environment

* Credo version (`mix credo -v`): 1.7.10-ref.main.35db3d57b+uncommittedchanges
* Erlang/Elixir version (`elixir -v`): 
    * Erlang/OTP 27 [erts-15.1.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]
    * Elixir 1.17.3 (compiled with Erlang/OTP 27)
* Operating system: Mac OS Sonoma 14.7.2

### What were you trying to do?

Run `mix credo`with the following code:

```elixir
anon_func.(Enum.random(1..10))
```

### Expected outcome

Tell me my code is ok because Enum.random(1..10) _is_ used inside the anonymous function call

### Actual outcome

Error message produced:

> There should be no unused return values for Enum functions

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest version of Credo.

If it is not, please feel free to re-open this issue!
